src/gwc/src/main/java/org/geoserver/gwc/dispatch/GwcServiceProxy.java

```diff
@@ -148,11 +151,16 @@ public void verifyAccess(HttpServletRequest rawRequest) throws org.geotools.ows.
             
         } else if (rawRequest.getPathInfo().toLowerCase().startsWith("/service/wmts")) {
             String layer = (String) parameters.get("LAYER");
-                        
+
             if (layer != null) {
-                GWC.get().verifyAccessLayer(layer, null);
+                String tileMatrix = (String) parameters.get("TileMatrix");
+                String srs = tileMatrix.substring(0, tileMatrix.lastIndexOf(":"));
+                int level = Integer.parseInt(tileMatrix.substring(tileMatrix.lastIndexOf(":") + 1));
```

Comment:
Rather than parsing out the level yourself, maybe try using the same logic as the WMTS implementation here: https://github.com/GeoWebCache/geowebcache/blob/master/geowebcache/wmts/src/main/java/org/geowebcache/service/wmts/WMTSService.java#L238